eva_submission/eload_brokering.py

```diff
@@ -58,19 +59,24 @@ def broker_to_ena(self, force=False):
             # Upload XML to ENA
             ena_uploader.upload_xml_files_to_ena(submission_file, project_file, analysis_file)
             self.eload_cfg.set('brokering', 'ena', value=ena_uploader.results)
+            self.eload_cfg.set('brokering', 'ena', 'date', value=self.now)
+            self.eload_cfg.set('brokering', 'ena', 'pass', value=not bool(ena_uploader.results['error']))
         else:
             self.info('Brokering to ENA has already been run, Skip!')
 
     def upload_to_bioSamples(self, force=False):
         metadata_spreadsheet = self.eload_cfg['validation']['valid']['metadata_spreadsheet']
         sample_tab_submitter = SampleMetadataSubmitter(metadata_spreadsheet)
-        if not (sample_tab_submitter.check_submit_done() or self.eload_cfg.query('brokering', 'Biosamples', 'Samples'))\
-                or force:
+        if sample_tab_submitter.check_submit_done() and not force:
+            self.info('Biosamples accession already provided in the metadata, Skip!')
+            self.eload_cfg.set('brokering', 'Biosamples', 'pass', value=True)
+        elif not self.eload_cfg.query('brokering', 'Biosamples', 'Samples') and not force:
```

Comment:
again, i think this logic is wrong, or else my boolean brain is on backwards today
```suggestion
        elif self.eload_cfg.query('brokering', 'Biosamples', 'Samples') and not force:
```

Reply:
either way, it might be worth writing some tests to cover these sorts of issues.

Reply:
Very good point. I'll add test for this

Reply:
Tests are done